config/routes.rb

```diff
@@ -27,7 +27,11 @@
         get "/find_all", to: "find#index"
         get "/random", to: "random#show"
       end
-      resources :transactions, only: [:show, :index]
+      resources :transactions, only: [:show, :index] do
+        scope module: :transactions do
+          get "/invoice", to: "invoice#show"
```

Comment:
Ah - I see what you're saying.  This looks like a good way to handle it.  Nice find.